Sources/CMakeLists.txt

```diff
@@ -12,12 +12,28 @@
 ##
 ##===----------------------------------------------------------------------===##
 
-add_subdirectory(_FoundationCShims)
-
-# Disable the macro build on Windows until we can correctly build it for the host architecture
-if(NOT CMAKE_SYSTEM_NAME STREQUAL Windows)
-	add_subdirectory(FoundationMacros)
+include(ExternalProject)
+if(CMAKE_HOST_WIN32)
+  set(_FoundationMacrosSwiftFlags "-use-ld=lld")
+endif()
+ExternalProject_Add(FoundationMacros
+  SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/FoundationMacros"
+  PREFIX "${CMAKE_BINARY_DIR}/_deps"
+  BINARY_DIR "macros"
+  CMAKE_ARGS
+    -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
+    -DCMAKE_SYSTEM_NAME=${CMAKE_HOST_SYSTEM_NAME}
+    "-DCMAKE_Swift_FLAGS=${_FoundationMacrosSwiftFlags} ${CMAKE_HOST_Swift_FLAGS}"
+  INSTALL_COMMAND "")
+ExternalProject_Get_Property(FoundationMacros BINARY_DIR)
+if(CMAKE_HOST_WIN32)
+  set(_SwiftFoundation_PredicateMacro "${BINARY_DIR}/FoundationMacros.dll#PredicateMacro")
+  set(_SwiftFoundation_ExpressionMacro "${BINARY_DIR}/FoundationMacros.dll#ExpressionMacro")
+else()
+  set(_SwiftFoundation_PredicateMacro "${BINARY_DIR}/FoundationMacros#PredicateMacro")
+  set(_SwiftFoundation_ExpressionMacro "${BINARY_DIR}/FoundationMacros#ExpressionMacro")
```

Comment:
I think I'm still confused on what the intention of these settings are? Are they supposed to be paths that we provide as `-plugin-path` to the FoundationEssentials build? If so, I think I'd expect them to be in the library directory on non-windows and also look like `libFoundationMacros.so` rather than `FoundationMacros#ExpressionMacro`.

Reply:
Yes, this was the expected parameter for the macro. Although, after the discussion with @rintaro yesterday, I don't think it matters. Once we have the custom option for the macro, this will no longer be needed.